src/nvim/ex_docmd.c

```diff
@@ -9952,3 +9953,22 @@ bool cmd_can_preview(char_u *cmd)
 
   return false;
 }
+
+int n_get_commands( void )
+{
+  return ucmds.ga_len;
+}
+
+Dictionary get_dict_command(int i) {
```

Comment:
I think you can have the function return an Array directly, then separate `n_get_commands` is not needed.